pkg/object/pipeline/pipeline.go

```diff
@@ -0,0 +1,207 @@
+/*
+ * Copyright (c) 2017, MegaEase
+ * All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package pipeline
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/megaease/easegress/pkg/context"
+	"github.com/megaease/easegress/pkg/logger"
+	"github.com/megaease/easegress/pkg/supervisor"
+)
+
+const (
+	// Category is the category of Pipeline.
+	Category = supervisor.CategoryBusinessController
+
+	// Kind is the kind of Pipeline.
+	Kind = "Pipeline"
+)
+
+func init() {
+	supervisor.Register(&Pipeline{})
+}
+
+type (
+	// Pipeline is general pipeline of Easegress
+	Pipeline struct {
+		superSpec      *supervisor.Spec
+		spec           *Spec
+		runningFilters []*runningFilter
+	}
+
+	runningFilter struct {
+		spec       *FilterSpec
+		rootFilter Filter
+		filter     Filter
+	}
```

Comment:
propose to rename `runningFilter`, maybe `filterInstance`, the name always confused me even in the HTTP pipeline.

Reply:
the `rootFilter` is the one we registered to pipeline and use it to create the running filter. So, maybe another name here, like `filterTemplate`?

Reply:
I will rm it since it is not used.